core/src/main/java/com/datastax/oss/driver/internal/core/connection/DecorrelatedJitterReconnectionPolicy.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.datastax.oss.driver.internal.core.connection;
+
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.util.Random;
+import net.jcip.annotations.ThreadSafe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A reconnection policy, similar to {@link ExponentialReconnectionPolicy}, that adds jitter to the
+ * delay to avoid cases where lots of clients all try to reconnect at the same exponential back-off
+ * intervals.
+ *
+ * <p>It uses the same schedule implementation for individual nodes or the control connection.
+ * Unlike {@link ExponentialReconnectionPolicy}, the number of attempts do not factor into the delay
+ * calculation. The implementation here uses a multiple of the previously calculated delay as the
+ * upper bound of a randomly generated delay. This results in early reconnection attempts being
+ * close to, but not always exactly, the base delay. As attempts build up, the upper bound on the
+ * delay will generally grow, allowing for longer and longer delays. However, since the delay uses a
+ * random number between that upper bound and the base delay, it is possibly for later attempts to
+ * use a delay that is close to (or even as low as) the base delay. The general algorithm is:
+ *
+ * <p>{@code delay = Random(base-delay, previous-delay*3)}
+ *
+ * <p>To activate this policy, modify the {@code advanced.reconnection-policy} section in the driver
+ * configuration, for example:
+ *
+ * <pre>
+ * datastax-java-driver {
+ *   advanced.reconnection-policy {
+ *     class = DecorrelatedJitterReconnectionPolicy
+ *     base-delay = 1 second
+ *     max-delay = 60 seconds
+ *   }
+ * }
+ * </pre>
+ *
+ * See {@code reference.conf} (in the manual or core driver JAR) for more details.
+ */
+@ThreadSafe
+public class DecorrelatedJitterReconnectionPolicy extends ExponentialReconnectionPolicy {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(DecorrelatedJitterReconnectionPolicy.class);
+
+  private final String logPrefix;
+  private final Random jitterRandom = new Random();
```

Comment:
Isn't there a need to use ThreadLocalRandom?

Reply:
I have changed the implementation and replaced `Random` with ThreadLocalRandom`. Good catch.